fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingThrottle.java

```diff
@@ -196,20 +196,22 @@ <R> CompletableFuture<R> throttledRunAsync(@Nonnull final Function<FDBRecordStor
 
         AtomicInteger tries = new AtomicInteger(0);
         CompletableFuture<R> ret = new CompletableFuture<>();
-        final ExponentialDelay delay = new ExponentialDelay(common.getRunner().getDatabase().getFactory().getInitialDelayMillis(),
-                common.getRunner().getDatabase().getFactory().getMaxDelayMillis());
         AsyncUtil.whileTrue(() -> {
             loadConfig();
             return common.getRunner().runAsync(context -> common.getRecordStoreBuilder().copyBuilder().setContext(context).openAsync().thenCompose(store -> {
-                for (Index index: common.getTargetIndexes()) {
-                    IndexState indexState = store.getIndexState(index);
-                    if (indexState != expectedIndexState) {
-                        throw new RecordCoreStorageException("Unexpected index state",
-                                LogMessageKeys.INDEX_NAME, index.getName(),
-                                common.getRecordStoreBuilder().getSubspaceProvider().logKey(), common.getRecordStoreBuilder().getSubspaceProvider().toString(context),
-                                LogMessageKeys.INDEX_STATE, indexState,
-                                LogMessageKeys.INDEX_STATE_PRECONDITION, expectedIndexState);
+                List<IndexState> indexStates = common.getTargetIndexes().stream().map(store::getIndexState).collect(Collectors.toList());
+                if (indexStates.stream().anyMatch(state -> state != expectedIndexState)) {
+                    if (indexStates.stream().allMatch(state -> state == IndexState.READABLE)) {
+                        throw new IndexingBase.UnexpectedReadableException(true, "All indexes are built");
                     }
+                    if (indexStates.stream().allMatch(state -> state == expectedIndexState || state == IndexState.READABLE)) {
+                        throw new IndexingBase.UnexpectedReadableException(false, "Some indexes are built");
+                    }
```

Comment:
I'm not sure I understand this condition. Should this just be:

```java
indexStates.stream().anyMatch(state -> state == IndexState.READABLE)
```

That would seem to more closely align with "Some indexes are built"

Reply:
Here there are 3 possible exceptions (in a short circuit order): 
1. All the indexes are now readable.
2. Some indexes are built, but all the others are in the expected state.
3. Some indexes are not in the expected state (disabled?). 
For the non mutual indexer, each one of these exception should be handled differently.

Reply:
Oh, I see. It might be helpful to add a comment to that affect--that this is to handle the case that some indexes are built, and all of the rest are in the expected state, and that the mutual indexer handles that differently than other unexpected state conditions.

Reply:
✅